WordPress/Sniffs/Theme/NonPrintableCheckSniff.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Restricts the use of non-printable characters in themes
+ *
+ * @link    https://make.wordpress.org/themes/handbook/review/required/theme-check-plugin/#info
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_NonPrintableCheckSniff extends WordPress_Sniff {
+
+	/**
+	 * A list of tokenizers this sniff supports.
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array(
+		'PHP',
+		'CSS',
+		'JS',
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return ( array_merge( PHP_CodeSniffer_Tokens::$commentTokens, PHP_CodeSniffer_Tokens::$emptyTokens, PHP_CodeSniffer_Tokens::$stringTokens, array(
```

Comment:
@jrfnl Yes, my assumption was that the non-printable chars can be sneaked in those kind of tokens without breaking the linting, so my best guess was the selected categories. 
I have also added the T_BAD_CHARCTER but i saw that is breaking the build on PHP7.